mantidproject/mantid
# New algorithm MaskBinsIf

**Description of work.**

This adds a new convenience algorithm for masking bins with criteria specified as muparser expression.

**To test:**

```
CreateSampleWorkspace(BankPixelWidth=100, NumBanks=1, OutputWorkspace='out')
MaskBinsIf(InputWorkspace='out', Criterion='s>10 && s<20 && x>1000 && x<2000', OutputWorkspace='out')
```

Open the matrix view of the workspace and check that the greyed bins are as expected. 
Play around with the criteria after having read the documentation.



Fixes #23745 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
